File: app/src/main/java/com/pij/zworkout/list/viewmodel/ShowDetailFeature.java
package com.pij.zworkout.list.viewmodel;

import android.support.annotation.NonNull;

import com.annimon.stream.Optional;
import com.pij.horrocks.Interaction;
import com.pij.horrocks.Reducer;
import com.pij.zworkout.list.Model;
import com.pij.zworkout.list.WorkoutInfo;

/**
 * <p>Created on 01/03/2018.</p>
 *
 * @author Pierrejean
 */

public class ShowDetailFeature implements Interaction<WorkoutInfo, Model> {
    @NonNull
    @Override
    public Reducer<Model> process(@NonNull WorkoutInfo workoutInfo) {
        return current -> current.toBuilder()
                .showWorkout(Optional.of(workoutInfo))
                .build();
    }
}

File: app/src/main/java/com/pij/zworkout/list/feature/ShowDetailFeature.kt
package com.pij.zworkout.list.feature

import com.pij.horrocks.Interaction
import com.pij.horrocks.Reducer
import com.pij.zworkout.list.Model
import com.pij.zworkout.list.WorkoutInfo

/**
 *
 * Created on 01/03/2018.
 *
 * @author Pierrejean
 */

class ShowDetailFeature : Interaction<WorkoutInfo, Model> {

    override fun process(workoutInfo: WorkoutInfo): Reducer<Model> {
        return Reducer { it.copy(showWorkout = workoutInfo) }
    }
}
